src/Adapter/Configuration.php

```diff
@@ -345,39 +353,43 @@ private function getLocalized($key, ?int $shopId = null, ?int $shopGroupId = nul
     }
 
     /**
-     * @param ShopConstraint|null $shopConstraint
+     * @param ShopConstraint $shopConstraint
      *
      * @return int|null
      */
-    private function getShopId(?ShopConstraint $shopConstraint): ?int
+    private function getShopId(ShopConstraint $shopConstraint): ?int
     {
-        return null !== $shopConstraint && null !== $shopConstraint->getShopId()
+        return null !== $shopConstraint->getShopId()
             ? $shopConstraint->getShopId()->getValue()
             : null
         ;
     }
 
     /**
-     * @param ShopConstraint|null $shopConstraint
+     * @param ShopConstraint $shopConstraint
      *
      * @return int|null
      */
-    private function getShopGroupId(?ShopConstraint $shopConstraint): ?int
+    private function getShopGroupId(ShopConstraint $shopConstraint): ?int
     {
-        return null !== $shopConstraint && null !== $shopConstraint->getShopGroupId()
-            ? $shopConstraint->getShopGroupId()->getValue()
-            : null
-        ;
-    }
+        if (null !== $shopConstraint->getShopGroupId()) {
+            return $shopConstraint->getShopGroupId()->getValue();
+        } elseif (null !== $shopConstraint->getShopId()) {
+            $shopGroupId = Shop::getGroupFromShop((int) $shopConstraint->getShopId()->getValue(), true);
+            // $shopGroupId can must not be false, it would mean that the shop group was not found for the given shop
+            if ($shopGroupId === false) {
+                throw new ShopException(
+                    sprintf(
+                        'Shop group with id %s was not found.',
+                        $shopGroupId
```

Comment:
`$shopGroupId` is always false in this case, so the exception message won't bring much info
```suggestion
                        'Shop group was not found for the shop with id %d.',
                        $shopConstraint->getShopId()->getValue()
```